src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -54,7 +54,13 @@ protected Calendar initialValue() {
     }
   };
 
-  public static Schema convertSchema(String tableName, ResultSetMetaData metadata)
+  private final boolean numericPrecisionMapping;
+
+  DataConverter(JdbcSourceTaskConfig config) {
+    this.numericPrecisionMapping = config.getBoolean(JdbcSourceConnectorConfig.NUMERIC_PRECISION_MAPPING_CONFIG);
```

Comment:
I had this dilemma too, and spent a few minutes thinking how to make it both elegant and static.
But then I checked that the class is only needed from at max one object - that is only one instance is ever going to be created I decided to go with the instance field.
Later on it could be generified so that it's more like a pluggable strategy.